examples/joystick/Joystick.cpp

```diff
@@ -16,145 +16,191 @@ struct JoystickObject
 {
     sf::Text label;
     sf::Text value;
+
+    void draw(sf::RenderWindow& window, const sf::RenderStates& states = sf::RenderStates::Default) const
+    {
+        window.draw(label, states);
+        window.draw(value, states);
+    }
 };
 
 using Texts = std::unordered_map<std::string, JoystickObject>;
-Texts              texts;
-std::ostringstream sstr;
-float              threshold = 0.1f;
+
+auto getStrm()
+{
+    std::ostringstream sstr;
+    // Set up our string conversion parameters
+    sstr.precision(2);
+    sstr.setf(std::ios::fixed | std::ios::boolalpha);
+    return sstr;
+}
 
 // Axes labels in as strings
 const std::array<std::string, 8> axislabels = {"X", "Y", "Z", "R", "U", "V", "PovX", "PovY"};
 
-// Helper to set text entries to a specified value
-template <typename T>
-void set(const std::string& label, const T& value)
+struct JoystickDisplay
 {
-    sstr.str("");
-    sstr << value;
-    texts.at(label).value.setString(sstr.str());
-}
+    unsigned int index   = 0;
+    bool         present = false;
+    Texts        texts;
 
-// Update joystick identification
-void updateIdentification(unsigned int index)
-{
-    sstr.str("");
-    sstr << "Joystick " << index << ":";
-    auto& [label, value] = texts.at("ID");
-    label.setString(sstr.str());
-    value.setString(sf::Joystick::getIdentification(index).name);
-}
 
-// Update joystick axes
-void updateAxes(unsigned int index)
-{
-    for (unsigned int j = 0; j < sf::Joystick::AxisCount; ++j)
+    void init(unsigned int idx, const sf::Font& font)
```

Comment:
Why an `init` function instead of a constructor?

Reply:
Added 2 constructors.